go/cobradocs_sync.go

```diff
@@ -0,0 +1,109 @@
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/google/go-github/v53/github"
+	"github.com/pkg/errors"
+	"github.com/vitess.io/vitess-bot/go/git"
+	"github.com/vitess.io/vitess-bot/go/shell"
+)
+
+// synchronize cobradocs from main and release branches
+func synchronizeCobraDocs(
+	ctx context.Context,
+	client *github.Client,
+	vitess *git.Repo,
+	website *git.Repo,
+	pr *github.PullRequest,
+	prInfo prInformation,
+) (*github.PullRequest, error) {
+	op := "update cobradocs"
+	branch := "prod"
+
+	websiteProdRef, _, err := client.Git.GetRef(ctx, website.Owner, website.Name, "heads/"+branch)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to fetch prod ref for repository %s/%s to %s on Pull Request %d", website.Owner, website.Name, op, prInfo.num)
+	}
+
+	newBranch := fmt.Sprintf("synchronize-cobradocs-for-%d", pr.GetNumber())
+	_, err = website.CreateBranch(ctx, client, websiteProdRef, newBranch)
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to create git ref %s ref for repository %s/%s to %s on Pull Request %d", newBranch, website.Owner, website.Name, op, prInfo.num)
+	}
+
+	if err := setupRepo(ctx, vitess, prInfo, op); err != nil {
+		return nil, err
+	}
+
+	if err := vitess.FetchRef(ctx, "origin", "--tags"); err != nil {
+		return nil, errors.Wrapf(err, "Failed to fetch tags in repository %s/%s to %s on Pull Request %d", vitess.Owner, vitess.Name, op, prInfo.num)
+	}
+
+	if err := setupRepo(ctx, website, prInfo, op); err != nil {
+		return nil, err
+	}
+
+	// Checkout the new branch we created.
+	if err := website.Checkout(ctx, newBranch); err != nil {
+		return nil, errors.Wrapf(err, "Failed to checkout repository %s/%s to branch %s to %s on Pull Request %d", website.Owner, website.Name, newBranch, op, prInfo.num)
+	}
+
+	// Run the sync script (which authors the commit already).
+	_, err = shell.NewContext(ctx, "./tools/sync_cobradocs.sh").InDir(website.LocalDir).WithExtraEnv(
+		fmt.Sprintf("VITESS_DIR=%s", vitess.LocalDir),
+		"COBRADOCS_SYNC_PERSIST=yes",
+		// `COBRADOC_VERSION_PAIRS="main:19.0"`, // here until rc becomes a real tag
+	).Output()
+	if err != nil {
+		return nil, errors.Wrapf(err, "Failed to run cobradoc sync script in repository %s/%s to %s on Pull Request %d", website.Owner, website.Name, newBranch, prInfo.num)
+	}
+
+	// TODO: do we need to amend the commit to change the author to the bot?
+
```

Comment:
If i recall correctly, I think I had to amend the commit when doing a cherry-pick / create a new commit.

Reply:
i meant this more in the sense that the `./tools/sync_cobradocs.sh` script in `website` authors a commit, and i'm not sure if, when running in prod, that will author as the bot user we want or not (nor am i sure how to test that without "just seeing what happens" haha). if it's authoring as the wrong user, then we'll need to amend that commit here to set the author/committer correctly

Reply:
Oh right. We can amend once we merge that in prod and see if it works :)